### ISSUE TITLE
XQuartz dependency when installing finch via `pidgin --without-GUI`

### ISSUE THREAD

**Issue by sweenzor** [Homebrew/homebrew#37919]
From @cakeface:
> l'm now getting a dependency on XQuartz after switching from the finch formula to using pidgin --without-GUI. I don't feel like I should be relying on X for finch installs and I don't really want to have to install XQuartz to use finch.

I'm also experiencing this issue:
```
> brew install pidgin --without-GUI
cairo: XQuartz is required to install this formula.
```

Related (I think) to #35043 /cc @DomT4

**Comment by DomT4** [comment]
This looks like the *super-fun* problem where ` cairo ` gets dragged in as a dep of a dep, and because ` cairo ` is built with ` x11 ` by default other formulae expect ` x11 ` consequently. Looking into it.

**Comment by sweenzor** [comment]
:+1:

**Comment by DomT4** [comment]
Yup. That's the issue. Working on a fix now. Just need to rejiggle the deps around a bit to make the deps of deps more explicit really, and remove `x11` from those deps of deps.

**Comment by cakeface** [comment]
Thanks @DomT4! Transitive dependencies are a joy!

**Comment by DomT4** [comment]
Fixed in #37922.
